pkg/cmd/server/crypto/crypto.go

```diff
@@ -472,7 +479,17 @@ func newSigningCertificateTemplate(subject pkix.Name) (*x509.Certificate, error)
 }
 
 // Can be used for ListenAndServeTLS
-func newServerCertificateTemplate(subject pkix.Name, hosts []string) (*x509.Certificate, error) {
+func newServerCertificateTemplate(subject pkix.Name, hosts []string, expireDays int) (*x509.Certificate, error) {
+
+	var lifetimeInDays = defaultCertificateLifetimeInDays
+	if expireDays > 0 {
+		lifetimeInDays = expireDays
+	}
+
+	warnAboutCertificateLifeTime(subject.CommonName, lifetimeInDays, defaultCertificateLifetimeInDays)
```

Comment:
don't hide the check inside this function, wrap this with the if and use the function just to print warning.